agent/doc/UML/hydra_lab_exception_monitor_design.puml

```diff
@@ -0,0 +1,62 @@
+@startuml exception_monitor
+interface ExceptionReporter {
+    + void reportException(Exception e)
+    + void reportException(Exception e ,Thread thread)
+    + void reportException(String message, StackTraceElement[] stackTrace, String threadName)
+}
+
+class ExceptionReporterManager {
+    List<ExceptionReporter> reporters
+    + void registerExceptionReporter(ExceptionReporter reporter)
+    + void reportException(Exception e)
+    + void reportException(Exception e ,Thread thread)
+    + void reportException(String message, StackTraceElement[] stackTrace, String threadName)
+}
+
+class AppCenterReporter implements ExceptionReporter {
+    AppCenterClient appCenterClient;
+
+    + void reportException(Exception e)
+    + void reportException(Exception e ,Thread thread)
+    + void reportException(String message, StackTraceElement[] stackTrace, String threadName)
+}
+
+
+class FileErrorLogger implements ExceptionReporter {
+    + void reportException(Exception e)
+    + void reportException(Exception e ,Thread thread)
+    + void reportException(String message, StackTraceElement[] stackTrace, String threadName)
+    + void writeToFile(String message, StackTraceElement[] stackTrace)
+}
+
+class AppCenterClient{
+    public HandledErrorLog createErrorLog(Thread thread, Exception testException, boolean fatal)
+}
+
+AppCenterClient <-- AppCenterReporter : call
+
+abstract class ch.qos.logback.core.AppenderBase<E> {
+    + abstract void append(E var1);
+}
+
+class LogbackExceptionHandler extends ch.qos.logback.core.AppenderBase {
+    ExceptionReporterManager exceptionReporterManager;
+    + void append(ILoggingEvent event)
+}
+
+class ThreadExceptionHandler implements Thread.UncaughtExceptionHandler {
```

Comment:
Do we have a exception listener for the thread pool?

Reply:
Currently, we have 2 types of Thread pool, we can customize a ThreadFactory class to init them.
![image](https://github.com/microsoft/HydraLab/assets/26757995/26773cad-f388-498b-bd12-6602ebd17cf9)